src/Plugin/VersionBridgePlugin.php

```diff
@@ -0,0 +1,21 @@
+<?php
+
+namespace Http\Client\Common\Plugin;
+
+use Psr\Http\Message\RequestInterface;
+
+/**
+ * A plugin that helps you migrate from php-http/client-common 1.x to 2.x. This
+ * will also help you to support PHP5 at the same time you support 2.x.
+ *
+ * @author Tobias Nyholm <tobias.nyholm@gmail.com>
+ */
+trait VersionBridgePlugin
+{
+    abstract protected function doHandleRequest(RequestInterface $request, callable $next, callable $first);
+
+    public function handleRequest(RequestInterface $request, callable $next, callable $first)
```

Comment:
Isn't this missing the return type hint?

Reply:
This is for master (ie 1.9.0). 

Here we should be compatible with httplug 1. Check #133 for httplug 2.0.

Reply:
Ah, I didn't see that PR yet. 👍